### PR TITLE
Add flag to open docs:  x.py doc --open

### PR DESCRIPTION
**Pull request by dtolnay**
This aligns with Cargo's flag `cargo doc --open`.

Tested with:

```bash
   # opens doc/index.html
x.py doc --stage 0 --open
x.py doc --stage 0 --open src/doc

   # opens doc/book/index.html
x.py doc --stage 0 --open src/doc/book

   # opens doc/std/index.html
x.py doc --stage 0 --open src/libstd

   # opens doc/proc_macro/index.html
x.py doc --stage 0 --open src/libproc_macro

   # opens both
x.py doc --stage 0 --open src/libstd src/libproc_macro
```

### CODE REVIEW

**Review by jyn514 — APPROVED** [review]
Just this week I tried to use this and was surprised it didn't exist :)

File: src/bootstrap/doc.rs [hunk 1]
```diff
@@ -70,6 +70,35 @@ book!(
     RustdocBook, "src/doc/rustdoc", "rustdoc";
 );
 
+fn open(builder: &Builder<'_>, path: impl AsRef<Path>) {
+    if builder.config.dry_run || !builder.config.cmd.open() {
+        return;
+    }
+
+    let path = path.as_ref();
+    builder.info(&format!("Opening doc {}", path.to_string_lossy()));
```
Comment (Mark-Simulacrum) [inline src/bootstrap/doc.rs:79]: nit: `.display()`?

File: src/bootstrap/doc.rs [hunk 2]
```diff
@@ -70,6 +70,35 @@ book!(
...
+
+    // ignore error
+    let _ = opener::open(path);
```
Comment (Mark-Simulacrum) [inline src/bootstrap/doc.rs:82]: I would have expected us to print the error here - at least on higher levels of verbosity, is there a reason we're ignoring it? I imagine it'd be useful for people who try to find out why open doesn't work for them.

File: src/bootstrap/doc.rs (hunk 1 above)
Comment (dtolnay) [reply src/bootstrap/doc.rs]: Fixed.

File: src/bootstrap/doc.rs (hunk 2 above)
Comment (dtolnay) [reply src/bootstrap/doc.rs]: Done. On my machine the error looks like:

```console
    Finished release [optimized] target(s) in 0.29s
Opening doc /git/rust/build/x86_64-unknown-linux-gnu/doc/std/index.html
command 'xdg-open (internal)' did not execute successfully; exit code: 4
command stderr:
gio: file:///git/rust/build/x86_64-unknown-linux-gnu/doc/std/index.html: Error when getting information for file “/git/rust/build/x86_64-unknown-linux-gnu/doc/std/index.html”: No such file or directory

Build completed successfully in 0:00:08
```

**Comment by Mark-Simulacrum** [comment]
@bors r+

Thanks!

**Pull request merged**